src/libstd/sys/unix/process.rs

```diff
@@ -311,6 +312,18 @@ impl Process {
         if !envp.is_null() {
             *sys::os::environ() = envp as *const _;
         }
+
+        // Reset signal handling so the child process starts in a
+        // standardized state. We ignore SIGPIPE, and signal-handling
+        // libraries often set a mask; both of these get inherited,
+        // which most UNIX programs don't expect.
+        let mut set: c::sigset_t = mem::uninitialized();
+        if c::sigemptyset(&mut set) != 0 ||
+           c::pthread_sigmask(c::SIG_SETMASK, &set, ptr::null_mut()) != 0 ||
+           c::signal(libc::SIGPIPE, c::SIG_DFL) == c::SIG_ERR {
+            fail(&mut output);
+        }
```

Comment:
> The old implementation used `sigprocmask` instead of `pthread_sigmask`

Yeah, either is fine here. `pthread_sigmask` is a strict superset of `sigprocmask`: it's defined to do the same thing if the process is single-threaded, and has well-defined behavior if the process is multithreaded, so I've just been making myself forget that `sigprocmask` exists. It's also one fewer thing to bind, which also reduces the temptation for a future commit to call it in a maybe-multithreaded context (e.g., if you wanted to mask signals around the call to fork, as libc `system` does). I guess there's an argument that it requires `-lpthread`, so if we expect to have UNIXish ports where we aren't using pthreads (???) we'd prefer `sigprocmask`, plus a warning comment about threads.

> The old implementation also did nothing with `SIGPIPE` or other signals

Yes, I think it's libstd's responsibility to unset any ignored-signal dispositions it sets, if that disposition is for internal use. `SIGPIPE` is the only one that's ignored; the other two signals libstd touches get handlers. (As it happens, I think there's also an argument that `SIGPIPE` is the only one it ever makes sense to ignore-but-not-inherit, since it's the only one with a corresponding `EPIPE`. If you're ignoring stuff like `SIGHUP`, you probably want the rest of your process group to ignore them too.)

Ignoring `SIGPIPE` is relatively recent (#13158, March 2014). The old `sigprocmask` code is much older (ea5dc54c3f0444fd3f20191fa1b1d94372c74c65, March 2011), and Rust was not setting _any_ signal handlers at that time.

> Could you clarify in this comment

Just the antecedent of "both of these"? I intend it as saying "`SIGPIPE` being ignored, and any signal mask", but I can rework that comment to be clearer, sure.